test/integration_test.rb

```diff
@@ -229,6 +236,51 @@ def test_mixed_borders
     end
     @workbook.apply_styles
     assert_equal 8, @workbook.styled_cells.count
-    assert_equal 6, @workbook.style_index.count
+    assert_equal 6, @workbook.styles.style_index.count
+    serialize(filename)
+  end
+
+  def test_merge_styles_1
+    filename = 'merge_styles_1'
+    @workbook.add_worksheet do |sheet|
+      sheet.add_row
+      sheet.add_row ['', '1', '2', '3']
+      sheet.add_row ['', '4', '5', '6']
+      sheet.add_row ['', '7', '8', '9']
+      sheet.add_style 'B2:D4', b: true
```

Comment:
Could you extract "bold" as a style, so that we test merging old styles and new styles (the same way as you do in the two tests that follow)?
``` ruby
bold = @workbook.styles.add_style b: true
# ...
sheet.add_row ['', '1', '2', '3'], style: [nil, bold, nil, nil]
```